Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -388,13 +391,49 @@ private void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun)
         if (!userImpulse || !TryComp<PhysicsComponent>(user, out var userPhysics))
             return;
 
+        RecoilKick(gun, (user, userPhysics), fromCoordinates, toCoordinates.Value);
+
         var shooterEv = new ShooterImpulseEvent();
         RaiseLocalEvent(user, ref shooterEv);
 
         if (shooterEv.Push || _gravity.IsWeightless(user, userPhysics))
             CauseImpulse(fromCoordinates, toCoordinates.Value, user, userPhysics);
```

Comment:
Recoil kick happens in the code next to the other recoil kick which makes shooting in zero-G move you.
This definitely needs something to disable one or the other when in zero-G since both shouldn't apply.

Also Resomi shooting in space means they go into an interplanetary orbit if we let the normal kick apply. Unsure if working as intended.